Client/Frontend/Browser/URLBarView.swift

```diff
@@ -620,8 +620,8 @@ extension URLBarView: TabLocationViewDelegate {
 
     func tabLocationViewDidTapLocation(tabLocationView: TabLocationView) {
         var locationText = delegate?.urlBarDisplayTextForURL(locationView.url)
-        if let host = locationView.url?.host {
-            locationText = locationView.url?.absoluteString?.stringByReplacingOccurrencesOfString(host, withString: host.asciiHostToUTF8())
+        if let text = locationText, let url = NSURL(string: text), let host = url.host {
```

Comment:
I'm having trouble following what this change is for. What does this fix? What's wrong with using `locationView.url`? Comments might be useful here to clear this up.